### PR TITLE
server.error.include-binding-errors does not recognize MethodValidationResult exceptions

### PR DESCRIPTION
**Pull request by quaff**
Fix GH-39858

### CODE REVIEW

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/error/DefaultErrorAttributes.java [hunk 1]
```diff
@@ -151,6 +157,11 @@ private void handleException(Map<String, Object> errorAttributes, Throwable erro
 				errorAttributes.put("errors", result.getAllErrors());
 			}
 		}
+		if (error instanceof MethodValidationResult result) {
+			if (result.hasErrors()) {
+				errorAttributes.put("errors", result.getAllErrors());
```
[the remark this answers is not in the document]
Comment (quaff) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/error/DefaultErrorAttributes.java]: done.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/error/DefaultErrorAttributes.java [hunk 2]
```diff
@@ -117,6 +119,14 @@ private String determineMessage(Throwable error, MergedAnnotation<ResponseStatus
 		if (error instanceof BindingResult) {
 			return error.getMessage();
 		}
+		if (error instanceof MethodValidationResult methodValidationResult) {
+			long errorCount = methodValidationResult.getAllErrors()
+				.stream()
+				.filter(ObjectError.class::isInstance)
+				.count();
+			return "Validation failed for method: %s, with %d %s".formatted(methodValidationResult.getMethod(),
+					errorCount, (errorCount > 2) ? "errors" : "error");
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/error/DefaultErrorAttributes.java:128]: `errorCount > 1`

Comment (quaff) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/error/DefaultErrorAttributes.java]: Sorry, updated.

**Comment by wilkinsona** [comment]
Thanks very much, @quaff.

**Pull request closed without merging**